packages/manager/modules/pci/src/components/project/storages/databases/text-accordion/controller.js

```diff
@@ -0,0 +1,39 @@
+export default class {
+  /* @ngInject */
+  constructor($timeout, $element, $window) {
+    this.$timeout = $timeout;
+    this.$element = $element;
+    this.$window = $window;
+  }
+
+  $onInit() {
+    this.hasLenghtyQuery = false;
+    this.expanded = false;
+    this.$timeout(() => {
+      this.createResizeObserver();
+      this.evalExpandButtonVisibility();
+    });
+  }
+
+  onResize() {
+    this.expanded = false;
+    this.$timeout(() => {
+      this.evalExpandButtonVisibility();
+    }, 100);
+  }
+
+  createResizeObserver() {
+    angular.element(this.$window).bind('resize', this.onResize.bind(this));
+  }
+
+  evalExpandButtonVisibility() {
+    const [textElement] = this.$element.find('div');
+    this.hasLenghtyQuery =
+      textElement.scrollWidth > textElement.offsetWidth ||
+      textElement.scrollHeight > textElement.offsetHeight;
```

Comment:
`textElement` could be `null` (I'm sure it is not, but it could be)
```suggestion
      textElement?.scrollWidth > textElement?.offsetWidth ||
      textElement?.scrollHeight > textElement?.offsetHeight;
```